app/kuma-dp/pkg/dataplane/accesslogs/server.go

```diff
@@ -37,58 +31,14 @@ func (s *accessLogServer) NeedLeaderElection() bool {
 func NewAccessLogServer(dataplane kumadp.Dataplane) *accessLogServer {
 	address := envoy.AccessLogSocketName(dataplane.Name, dataplane.Mesh)
 	return &accessLogServer{
-		server:     grpc.NewServer(),
-		newHandler: defaultHandler,
-		address:    address,
-	}
-}
-
-func (s *accessLogServer) StreamAccessLogs(stream envoy_accesslog.AccessLogService_StreamAccessLogsServer) (err error) {
-	// increment stream count
-	streamID := atomic.AddInt64(&s.streamCount, 1)
-
-	totalRequests := 0
-
-	log := logger.WithValues("streamID", streamID)
-	log.Info("starting a new Access Logs stream")
-	defer func() {
-		log.Info("Access Logs stream is terminated", "totalRequests", totalRequests)
-		if err != nil {
-			log.Error(err, "Access Logs stream terminated with an error")
-		}
-	}()
-
-	initialized := false
-	var handler logHandler
-	for {
-		msg, err := stream.Recv()
-		if err != nil {
-			if err == io.EOF {
-				return nil
-			}
-			return err
-		}
-
-		totalRequests++
-
-		if !initialized {
-			initialized = true
-
-			handler, err = s.newHandler(log, msg)
-			if err != nil {
-				return errors.Wrap(err, "failed to initialize Access Logs stream")
-			}
-			defer handler.Close()
-		}
-
-		if err := handler.Handle(msg); err != nil {
-			return err
-		}
+		server:  grpc.NewServer(),
+		address: address,
 	}
 }
 
 func (s *accessLogServer) Start(stop <-chan struct{}) error {
-	envoy_accesslog.RegisterAccessLogServiceServer(s.server, s)
+	v2.RegisterAccessLogServer(s.server)
+	v3.RegisterAccessLogServer(s.server)
```

Comment:
Interesting, now that the other PR has the `version` can weswitch back and forth between only 1 of these?

Reply:
Kuma DP will now register both ALS on one port.
Kuma CP will decide (using configurers which version of the API will be used)